### PR TITLE
Special game disable sciences from sending

### PR DESCRIPTION
**Pull request by PlayerNoon**
Brief description of the changes:

Adding feature to special games panel.
Allows to disable chosen science packs from sending. 
image

image

image

### Tested Changes:
- [x] I've tested the changes locally
- [ ] I've not tested the changes.

### CODE REVIEW

**Comment by TheBigZet** [comment]
How about displaying icons instead of names in setup gui?

Also, I'd prefer disabling the buttons instead of removing them (`enabled =false`), perhaps with some tooltip "disabled by special game"

**Comment by PlayerNoon** [comment]
review someone :D

**Comment by TheBigZet** [comment]
Ah I see what you did there. That feels overcomplicated. What I meant was to just disable the buttons in gui, like that:

**Comment by TheBigZet** [comment]
I wonder if inverted tickboxes in config gui wouldn't be more intuitive. It starts with all boxes ticked (every sci enabled) and to disable you have to un-tick a box. It would allow for less conversions in code too I think

File: comfy_panel/special_games.lua [hunk 1]
```diff
@@ -295,6 +317,34 @@ local function generate_limited_lives(lives_limit)
...
+		["utility-science-pack"] = packs[6],
+		["space-science-pack"] = packs[7]
+	}
+	local message = {"Special game generated. Disabled science:"}
+	for k, v in pairs(disabled_food) do
+		if v then
+			table.insert(message, Tables.food_long_to_short[k].short_name)
+		end
+	end
+	if table_size(message)>1 then
```
Comment (Ragnarok77-factorio) [inline comfy_panel/special_games.lua:337]: I hosted a game, one player joined a team, I enabled the event like that, yet I don't have the wrong message (should be purple disabled, not all science enabled) : 
image

**Review by Ragnarok77-factorio — CHANGES_REQUESTED** [review]
Disabling science doesnt work

File: comfy_panel/special_games.lua (hunk 1 above)
Comment (TheBigZet) [reply comfy_panel/special_games.lua:337]: Hmm it worked for me when I was testing it... Perhaps I had some uncomited changes

**Comment by Ragnarok77-factorio** [comment]
I was not here when it happened so couldn't check log factorio too but the report of the event where it happened once was : 
"went well no bug/issue, on tytd one guy went spectate when he realized he can't spam green".

**Review by Ragnarok77-factorio — APPROVED** [review]
OK, tested for months on main bb server, no issue

**Comment by Ragnarok77-factorio** [comment]
Fixed conflicts, I hope I didn't break anything, need to test just the disable sci + send to external server before merging.

**Comment by Ragnarok77-factorio** [comment]
I tested it, was fine, nothing else was really added, just removed some useless empty space that were added, and fixed a merge mistake.

**Review by Ragnarok77-factorio — APPROVED** [review]

**Pull request merged**